lynx/common/scene.py

```diff
@@ -11,13 +11,27 @@
 class Scene(Serializable):
     entities: List[Entity] = field(default_factory=list)
     _objects_map: Dict[Vector, Object] = field(default_factory=dict)
+    _id_map: Dict[int, Object] = field(default_factory=dict)
 
     def post_populate(self) -> NoReturn:
         for entity in self.entities:
             if type(entity) is Object:
                 self._objects_map[entity.position] = entity
+                self._id_map[entity.id] = entity
 
     def add_entity(self, entity: Entity) -> NoReturn:
         self.entities.append(entity)
         if type(entity) is Object:
             self._objects_map[entity.position] = entity
+            self._id_map[entity.id] = entity
+
+    def get_object_by_id(self, id: int) -> (Object | None):
```

Comment:
You can use `Union` return type from `typing` like e.g. `-> Union[Object, None]`.

Reply:
Is it wrong? I think I've seen people use sth like that.

Reply:
Nah. It's just that we already import `typing` in most of our files, so we may as well use it for consistency's sake. It's more Pythonic.
If you feel like it'a a _whatever_ thing then I'm not going to chase you to change it xD

Reply:
`-> (Object | None)` seems to be broken.

Reply:
Done